src/parser.cpp

```diff
@@ -2121,7 +2121,11 @@ namespace Sass {
       expr = parse_at_root_expression();
     }
     if (peek < exactly<'{'> >()) {
-      body = parse_block(true);
+      std::string feat("");
+      if (expr && expr->feature())
+        feat = expr->feature()->to_string();
+      // allow declarations in certain cases
+      body = parse_block(feat != "without");
```

Comment:
Unfortunately the `without` directive is too complex for this. For example should error.

```
a { b: c }

.d {
  @supports (foo: bar) {
    @media all {
      @at-root (without: media) {
        @extend a;
      }
    }
  }
}
```